scylla/src/transport/topology.rs

```diff
@@ -41,6 +43,52 @@ pub struct Peer {
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct Keyspace {
     pub strategy: Strategy,
+    pub tables: HashMap<String, Table>,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Table {
+    pub columns: HashMap<String, Column>,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Column {
+    pub type_: ColumnType,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct ColumnType {
+    pub cql_type: CqlType,
+    pub frozen: bool,
```

Comment:
Being frozen or not is relevant only for compound types: lists, sets, maps, tuples and user-defined types. Let's change the `CqlType` as follows:

```rust
pub enum CqlType {
    Native(NativeType),
    Compound {
        frozen: bool,
        type_: CompoundType,
    },
}
```

Reply:
Aren't tuples always frozen?

Reply:
Indeed they are, my mistake. How about using the following variants: `Native, Collection, Tuple, UserDefinedType` (Collection and UserDefinedType would have the frozen flag)?